### ISSUE TITLE
[Bug] Fix Flaky `TestEventsQueryClient_Subscribe_Succeeds` test

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#120]
## Objective

Fix flaky test preventing other PRs from merging to main.

## Origin Document

```bash
make itest 10 10 ./pkg/client/events_query/... -- -run=TestEventsQueryClient_Subscribe_Succeeds
```

![Screenshot 2023-10-31 at 2 25 47 PM](https://github.com/pokt-network/poktroll/assets/1892194/6b825bc2-107e-4912-ac2e-ecf29e8c98ff)

## Goals

- Unblock merges to main
- Improve flaky tests relates to the observable and downstream objects

## Deliverables

- [ ] A PR that fixes the test in the origin document
- [ ] Verification that `make itest 10 10 ./pkg/client/events_query/... -- -run=TestEventsQueryClient_Subscribe_Succeeds` works locally
- [ ] An improvement in CI 
- [ ] Remove the `t.Skip` message in `TestEventsQueryClient_Subscribe_Succeeds`

## Non-goals / Non-deliverables

- changes to the core business logic of the observable

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @bryanchriswhite  @h5law

**Comment by bryanchriswhite** [comment]
This is a really interesting case. :thinking: I'm only able to reproduce this one ~1/6000 times, so I guess @h5law wins. 🎰

**Comment by bryanchriswhite** [comment]
It seems like I was able to fix every other flaky test case **except this one** in #124. :sweat_smile:
